### PR TITLE
consolidate misc platform specific components

### PR DESCRIPTION
**Pull request by dagar**
Another small piece towards the platform reorganization that's been discussed. We still need to agree on a concrete plan for the vendor board layout, but for the time being we can continue to consolidate everything specific to a single platform in the relevant nuttx/posix/qurt directory. 

https://github.com/PX4/Firmware/pull/7806
https://github.com/PX4/Firmware/pull/8378

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Looks reasonable! Ideally one commit per OS to reduce rebase / merge drama potential, but I can live with it in the current form.

**Comment by dagar** [comment]
@davids5 yes something like that, but that's the bigger piece we need to settled on.

The oversimplified high level plan is...

 1. All platform specific pieces migrate to platorms/${OS}
 2. (undecided) All board specific pieces should go to a single location.

Personally what I want for 2 is a single folder that accounts for all pieces of a particular board.

For example px4fmu-v2 we have
   - nuttx-config
   - PX4 cmake config (nuttx_px4fmu-v2_default.cmake)
   - px4fmu-v2.prototype
   - src/drivers/boards/px4fmu-v2
   - anything else?

I want a single thing that defines the px4fmu-v2 and has all pieces. If we want to remove that board support it's trivial. If an adopter wants to create a slightly modified variant they can copy, paste, rename and go. 

Then there's the question of this living top level in a per vendor tree or within the platform support.

tl;dr - discuss on the next dev call?
